projects/sandbox/train/train/data/base.py

```diff
@@ -0,0 +1,516 @@
+        """
+        Size of strain tensor timeseries
+        the nerual network will analyze
+        """
+        return int(self.hparams.kernel_length * self.sample_rate)
+
+    @property
+    def num_params(self):
+        """Number of parameters inference is performed on"""
+        return len(self.hparams.inference_params)
+
+    @property
+    def val_batch_size(self):
+        """Use larger batch sizes when we don't need gradients."""
+        return int(4 * self.hparams.batch_size)
+
+    @property
+    def train_val_fnames(self):
+        """List of background files used for both training and validation"""
+        background_dir = self.data_dir / "train" / "background"
+        fnames = list(background_dir.glob("*.hdf5"))
+        return fnames
+
+    @property
+    def test_fnames(self):
+        """List of background files used for testing a trained model"""
+        test_dir = self.data_dir / "test" / "background"
+        fnames = list(test_dir.glob("*.hdf5"))
+        return fnames
+
+    def train_val_split(self) -> Sequence[str]:
+        """
+        Split background files into training and validation sets
+        based on the requested duration of the validation set
+        """
+        fnames = sorted(self.train_val_fnames)
+        durations = [int(fname.stem.split("-")[-1]) for fname in fnames]
+        valid_fnames = []
+        valid_duration = 0
+        while valid_duration < self.hparams.min_valid_duration:
+            fname, duration = fnames.pop(-1), durations.pop(-1)
+            valid_duration += duration
+            valid_fnames.append(str(fname))
+
+        train_fnames = fnames
+        return train_fnames, valid_fnames
+
+    # ================================================ #
+    # Helper utilities for preprocessing
+    # ================================================ #
+    def transform(self, parameters: Dict[str, Tensor]):
+        """
+        Make transforms to parameters before scaling
+        and performing training/inference.
+        For example, taking logarithm of hrss
+        """
+        return self.parameter_transformer(parameters)
+
+    def scale(self, parameters, reverse: bool = False):
+        """
+        Apply standard scaling to transformed parameters
+        """
+        parameters = parameters.transpose(1, 0)
+        scaled = self.standard_scaler(parameters, reverse=reverse)
+        scaled = scaled.transpose(1, 0)
+        return scaled
+
+    # ================================================ #
+    # Utilities for initial data loading and preparation
+    # ================================================ #
+    def setup(self, stage: str) -> None:
+        world_size, rank = self.get_world_size_and_rank()
+        self._logger = self.get_logger(world_size, rank)
+
+        # infer sample rate directly from background data
+        with h5py.File(self.train_fnames[0], "r") as f:
+            sample_rate = 1 / f[self.hparams.ifos[0]].attrs["dx"]
```

Comment:
This "dx" key is also a breaking change, and makes assumptions about the hdf5 dataset. This comment is tied to the one above.